onnx/defs/nn/defs.cc

```diff
@@ -685,7 +690,15 @@ Output case #2: Y (test mode)
     .TypeConstraint(
         "T",
         {"tensor(float16)", "tensor(float)", "tensor(double)"},
-        "Constrain input and output types to float tensors.");
+        "Constrain input and output types to float tensors.")
+    .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+        if (!ctx.getAttribute("is_test") ||
```

Comment:
Nit: Whether we have `is_test` or not, calling `propagateShapeAndTypeFromFirstInput` should be fine.